docs/Frequently Asked Questions/React Native/index.md

```diff
@@ -0,0 +1,244 @@
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SDLEventEmitter : RCTEventEmitter
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+###### SDLEventEmitter.m
+```objc
+#import "SDLEventEmitter.h"
+#import "ProxyManager.h"
+#import <React/RCTConvert.h>
+#import <SmartDeviceLink/SmartDeviceLink.h>
+
+@implementation SDLEventEmitter
+
+RCT_EXPORT_MODULE()
+
+- (instancetype)init {
+    self = [super init];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(getDoActionNotification:) name:<#Notification Name#> object:nil];
+
+    return self;
+}
+
+- (NSArray<NSString *> *)supportedEvents {
+    return @[@"DoAction"];
+}
+
+- (void)getDoActionNotification:(NSNotification *)notification {
+    if(self.sdlManager == nil) {
+        self.sdlManager = notification.userInfo[@"sdlManager"];
+    }
+    [self sendEventWithName:@"DoAction" body:@{@"type": @"actionType"}];
+}
+
+@end
+```
+
+##### Swift
+```swift
+@objc(SDLEventEmitter)
+class SDLEventEmitter: RCTEventEmitter {
+
+override init() {
+    NotificationCenter.default.addObserver(self, selector: #selector(doAction(_:)), name: Notification.Name(rawValue: "<#Notification Name#>", object: nil)
+    super.init()
+}
+
+@objc func doAction(_ notification: Notification) {
+    if self.sdlManger == nil {
+        self.sdlManager = notification.userInfo["sdlManager"]
+    }
+
+    sendEvent(withName: "DoAction", body: ["type": "actionType"])
+}
+
+override func supportedEvents() -> [String]! {
+    return ["DoAction"]
+}
+
+}
+```
+
+!!! NOTE
+Make sure you add `#import "React/RCTEventEmitter.h"` to the apps bridging header before moving forward if you're making a Swift application.
+!!!
+
+!!! NOTE
+If you're using Swift, you will need to create the Objective-C bridging class for `SDLEventEmitter` and add the proper `RCT_EXTERN_METHOD` wrapper.
+!!!
+
+```objc
+#import "React/RCTBridgeModule.h"
+#import "React/RCTEventEmitter.h"
+@interface RCT_EXTERN_MODULE(SDLEventEmitter, RCTEventEmitter)
+
+RCT_EXTERN_METHOD(eventCall:(eventCall: (id)dict))
+
+@end
+```
+
+The above example will then call into JavaScript with an event type `DoAction`. Inside your React Native(Javascript) code create a `NativeEventEmitter` object  within your `EventEmitter` module and add a listener for the event.
+
+```javascript
+import { NativeEventEmitter, NativeModules } from 'react-native';
+const  { SDLEventEmitter } = NativeModules;
+
+const testEventEmitter = new NativeEventEmitter(SDLEventEmitter);
+
+// Build a listener to listen for events
+const testData = testEventEmitter.addListener(
+    'DoAction',
+        () => SDLEventEmitter.eventCall({
+            "data": {
+                "low": "77",
+                "high": "87",
+                "currentTemp": "82",
+                "rain": "50%"
+            }
+        }   
+    )
+)
+```
+
+The last step is to wrap the method you wish to expose inside an `RCT_EXPORT_METHOD` for Objective-C and `RCT_EXTERN_METHOD` for Swift. 
```

Comment:
Doesn't say where the below code would go

Reply:
updated